base/commands/project/project_list_templates.go

```diff
@@ -0,0 +1,234 @@
+}
+
+func listTemplates(logger log.Logger, isLocal bool, isForce bool) ([]Template, error) {
+	sa := store.NewStoreAccessor(paths.Join(paths.Home(), storeFolder), logger)
+	if isLocal {
+		return listLocalTemplates()
+	}
+	if fetch, err := isFetch(sa); err != nil {
+		return []Template{}, err
+	} else if fetch || isForce {
+		ts, err := fetchTemplates()
+		if err != nil {
+			return nil, err
+		}
+		err = updateCache(sa, ts)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return listFromCache(sa)
+}
+
+func listLocalTemplates() ([]Template, error) {
+	var templates []Template
+	ts, err := paths.FindAll(paths.Templates(), func(basePath string, entry os.DirEntry) (ok bool) {
+		return entry.IsDir()
+	})
+	if err != nil {
+		return nil, err
+	}
+	for _, t := range ts {
+		templates = append(templates, Template{Name: t, Source: "local"})
+	}
+	return templates, nil
+}
+
+func fetchTemplates() ([]Template, error) {
+	var templates []Template
+	resp, err := http.Get(fetchURL())
+	if err != nil {
+		return []Template{}, err
+	}
+	respData, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return []Template{}, err
+	}
+	var data []map[string]any
+	err = json.Unmarshal(respData, &data)
+	if err != nil {
+		return []Template{}, err
+	}
+	for _, d := range data {
+		var tName string
+		var ok bool
+		if tName, ok = d["full_name"].(string); !ok {
+			return []Template{}, errors.New("error fetching repositories in the organization")
+		}
+		sName := strings.Split(tName, "/")
+		source := fmt.Sprintf("%s/%s", "github.com", sName[0])
+		name := sName[1]
+		templates = append(templates, Template{Name: name, Source: source})
+	}
+	return templates, nil
+}
+
+func updateNextFetchTime(sa *store.StoreAccessor) error {
+	_, err := sa.WithLock(func(s *store.Store) (any, error) {
+		return nil, s.SetEntry(bytes(nextFetchTimeKey),
+			bytes(strconv.FormatInt(time.Now().Add(cacheRefreshInterval).Unix(), 10)))
+	})
+	return err
+}
+
+func fetchURL() string {
+	s := strings.TrimPrefix(templateOrgURL(), "https://github.com/")
+	ss := strings.ReplaceAll(s, "/", "")
+	return fmt.Sprintf("https://api.github.com/users/%s/repos", ss)
+}
+
+func isFetch(s *store.StoreAccessor) (bool, error) {
+	entry, err := s.WithLock(func(s *store.Store) (any, error) {
+		return s.GetEntry(bytes(nextFetchTimeKey))
+	})
+	if err != nil {
+		if errors.Is(err, store.ErrKeyNotFound) {
+			return true, nil
+		}
+		return false, err
+	}
+	var fetchTS time.Time
+	t, err := strconv.ParseInt(string(entry.([]byte)), 10, 64)
+	if err != nil {
+		return false, err
+	}
+	fetchTS = time.Unix(t, 0)
+	if time.Now().After(fetchTS) {
+		return true, nil
+	}
+	return false, nil
+}
+
+func updateCache(sa *store.StoreAccessor, templates []Template) error {
+	b, err := json.Marshal(templates)
+	if err != nil {
+		return err
+	}
+	_, err = sa.WithLock(func(s *store.Store) (any, error) {
+		err = s.DeleteEntriesWithPrefix(templatesKey)
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Do we have to delete the template key?
Wouldn't setting the new value overwrite the older one?